scripts/trade0x-kovan.js

```diff
@@ -0,0 +1,275 @@
+const PayableProxyController = artifacts.require('PayableProxyController.sol')
+const Trade0x = artifacts.require('Trade0x')
+const PermitCallee = artifacts.require('PermitCallee')
+
+/**
+ * Kovan addresses
+ */
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000'
+const EXCHANGE_ADDR = '0xf1ec7d0ba42f15fb5c9e3adbe86431973e44764c'
+const ERC20PROXY_ADDR = '0xaa460127562482faa5df42f2c39a025cd4a1cc0a'
+const trade0xAddress = '0xF36e7676FaAaa07Ab92C6E4B6007cB65838F18d6'
+const otokenFactoryAddress = '0xb9D17Ab06e27f63d0FD75099d5874a194eE623e2'
+const payableProxyAddress = '0x5957A413f5Ac4Bcf2ba7c5c461a944b548ADB1A5'
+const controllerProxyAddress = '0xdEE7D0f8CcC0f7AC7e45Af454e5e7ec1552E8e4e'
+const marginPoolAddress = '0x8c7C60d766951c5C570bBb7065C993070061b795'
+const usdcAddress = '0xb7a4F3E9097C08dA09517b5aB877F7a917224ede'
+const wethAddress = '0xd0A1E359811322d97991E03f863a0C30C2cF029C'
+
+const ActionType = Object.freeze({
+  OpenVault: 0,
+  MintShortOption: 1,
+  BurnShortOption: 2,
+  DepositLongOption: 3,
+  WithdrawLongOption: 4,
+  DepositCollateral: 5,
+  WithdrawCollateral: 6,
+  SettleVault:7 ,
+  Redeem: 8,
+  Call: 9,
+})
+
+const Permit = [
+  {name: 'owner', type: 'address'},
+  {name: 'spender', type: 'address'},
+  {name: 'value', type: 'uint256'},
+  {name: 'nonce', type: 'uint256'},
+  {name: 'deadline', type: 'uint256'},
+]
+
+const expiryTime = new BigNumber(60 * 60 * 24)            // 1 day
+const usdcDecimals = 6
+const wethDecimals = 18
+
+const cmd = yargs
+  .option("makerPrivateKey", { describe: "0x Maker Private Key, wallet account[1]", type: "string", demandOption: true })
+  .argv;
+
+// create signer etherJS Object
+const makerEtherJS = new ethers.Wallet(cmd.makerPrivateKey)
+
+async function runExport() {
+  console.log("🔥🔥🔥 STARTING 0x TRADING SCRIPT 🔥🔥🔥")
+  console.log("0x order signer address: 🖋️ ", makerEtherJS.address)
+
+  let trade0xCallee = await Trade0x.at(trade0xAddress)
+  let otokenFactory = await OtokenFactory.at(otokenFactoryAddress)
+  let payableProxyController = await PayableProxyController.at(payableProxyAddress);
+  let controllerProxy = await Controller.at(controllerProxyAddress)
+  let marginPool = await MarginPool.at(marginPoolAddress);
+  let usdc = await ERC20.at(usdcAddress);
+  let weth = await WETH9.at(wethAddress);
+
+  const account = await web3.eth.getAccounts()
+  const taker = account[0]
+  const maker = account[1]
+
+  console.log("Taker address: ", taker)
+
+  if (makerEtherJS.address != maker) {
+    console.log("0x Maker address does not correspond to private key 🔴")
+    return;
+  }
+
+  const latestBlock = new BigNumber((await web3.eth.getBlock('latest')).timestamp);
+  let expiry = new BigNumber(latestBlock).plus(expiryTime)
+  let expiryDate = new Date(expiry.toNumber() * 1000)
+  expiryDate.setUTCHours(8)
+  expiryDate.setUTCMinutes(0)
+  expiryDate.setUTCSeconds(0)
+  expiry = new BigNumber(Math.floor(expiryDate.getTime() / 1000))
+
+  let callOption1;
```

Comment:
can we also test put options? :D

Reply:
IMO it is the same as the Trade0x is agnostic to what the option type is, it is the same asset/contract.